### PR TITLE
Retry (w/ back-off) downloading the installer binary

### PR DESCRIPTION
**Pull request by colemickens**
##### Description

- Replace `node-fetch` with node's builtin fetch
- Add `fetch-retry` as a new dep
- Use `fetch-retry` to retry and exponentially back-off on downloading the installer
- limit to 5 retries

Tested by running the GH CI workflow in my private repo, and it should run here again.

I don't do a lot of JS/TS and that's being generous to me. Please feel free to review harshly, happy to learn anything, even if nit-y. :pray: 

Additional context, per `fetch-retry` readme:
* it will only `retryOn` network failures by default
* (default is 3 retries, with static 1s delay, but we override, not the defaults, but the only invocation used)

##### Checklist

- [x] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

**Review by Hoverbear — APPROVED** [review]
Once the `any` use is resolved, I think it looks fine?

File: src/main.ts [hunk 1]
```diff
@@ -423,8 +436,14 @@ class NixInstallerAction {
       }
 
       if (response.body !== null) {
-        const streamPipeline = promisify(pipeline);
-        await streamPipeline(response.body, fs.createWriteStream(tempfile));
+        // shameless: https://stackoverflow.com/a/51302466
+        const fileStream = fs.createWriteStream(tempfile);
+        const responseBodyCast =
+          response.body as stream_web.ReadableStream<any>; // eslint-disable-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (colemickens) [reply src/main.ts]: Ack, I looked at this again and slightly have a better feel for what's going on. I can convert the fileStream be Web-y and then I don't have to cast, and don't have to use `any`.

**Comment by colemickens** [comment]
New commit, more node-y naming for `fetchRetry` and fixed the aforementioned casting/any issue.

**Comment by colemickens** [comment]
Huh. It's a bit tricky having `dist/` checked in, one has to be careful to ensure that it gets rebuilt after rebasing, and it probably means that it's "wrong" for inbetween commits. Huh.

**Comment by Hoverbear** [comment]
Yeah it's just ~great~.

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**